app/editor/components/LinkToolbar.tsx

```diff
@@ -31,39 +31,34 @@ function isActive(props: Props) {
   }
 }
 
-export default class LinkToolbar extends React.Component<Props> {
-  menuRef = React.createRef<HTMLDivElement>();
-
-  state = {
-    left: -1000,
-    top: undefined,
-  };
-
-  componentDidMount() {
-    window.addEventListener("mousedown", this.handleClickOutside);
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener("mousedown", this.handleClickOutside);
-  }
-
-  handleClickOutside = (event: Event) => {
-    if (
-      event.target instanceof HTMLElement &&
-      this.menuRef.current &&
-      this.menuRef.current.contains(event.target)
-    ) {
-      return;
-    }
-
-    this.props.onClose();
-  };
-
-  handleOnCreateLink = async (title: string) => {
-    const { dictionary, onCreateLink, view, onClose, onShowToast } = this.props;
+export default function LinkToolbar(props: Props) {
+  const menuRef = React.createRef<HTMLDivElement>();
+
+  const handleClickOutside = React.useCallback(
+    (event: Event) => {
+      if (
+        event.target instanceof HTMLElement &&
+        menuRef.current &&
+        menuRef.current.contains(event.target)
+      ) {
+        return;
+      }
+
+      props.onClose();
+    },
+    [menuRef, props]
+  );
+
+  React.useEffect(() => {
+    window.addEventListener("mousedown", handleClickOutside);
+    return () => window.removeEventListener("mousedown", handleClickOutside);
+  }, [handleClickOutside]);
+
+  const handleOnCreateLink = async (title: string) => {
```

Comment:
This seems to have broken the functionality `cmd`+`k` on an empty line no longer brings up the link insert toolbar. I suspect that these two callbacks need to be memoized with `useCallback`

Reply:
Just to be clear, are the two callbacks you've said `handleOnCreateLink` and `handleOnSelectLink`?

Reply:
Yes, anything that isn't memo'd